test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/ExtendedQueryTagTests.cs

```diff
@@ -47,25 +46,73 @@ public async Task GivenValidExtendedQueryTag_WhenGoThroughEndToEndScenario_ThenS
+            DicomTag patientAgeTag = DicomTag.PatientAge;
 
-            // upload file
-            string tagValue = "123";
-            DicomDataset dataset = Samples.CreateRandomInstanceDataset();
-            dataset.Add(tag, tagValue);
-            await _instanceManager.StoreAsync(new DicomFile(dataset));
-            InstanceIdentifier instanceId = dataset.ToInstanceIdentifier();
+            // Define DICOM files
+            DicomDataset instance1 = Samples.CreateRandomInstanceDataset();
+            DicomDataset instance2 = Samples.CreateRandomInstanceDataset();
+            DicomDataset instance3 = Samples.CreateRandomInstanceDataset();
 
-            // add tag
-            AddExtendedQueryTagEntry addExtendedQueryTagEntry = new AddExtendedQueryTagEntry { Path = tag.GetPath(), VR = tag.GetDefaultVR().Code, Level = QueryTagLevel.Instance };
-            var operationStatus = await _tagManager.AddTagsAsync(new[] { addExtendedQueryTagEntry });
+            // Annotate files
+            // (Disable Auto-validate)
+            instance1.Add(patientWeightTag, 68.0M);
+            instance2.Add(patientWeightTag, 50.0M);
+            instance3.Add(patientWeightTag, 81.0M);
+
+#pragma warning disable CS0618
+            instance1.AutoValidate = false;
+            instance2.AutoValidate = false;
+#pragma warning restore CS0618
+
+            instance1.Add(patientAgeTag, "foobar");
+            instance2.Add(patientAgeTag, "invalid");
+            instance3.Add(patientAgeTag, "053Y");
+
+            // Upload files (with a few errors)
+            await _instanceManager.StoreAsync(new DicomFile(instance1));
+            await _instanceManager.StoreAsync(new DicomFile(instance2));
+            await _instanceManager.StoreAsync(new DicomFile(instance3));
+
+            // Add extended query tags
+            var operationStatus = await _tagManager.AddTagsAsync(
+                new AddExtendedQueryTagEntry[]
+                {
+                    new AddExtendedQueryTagEntry { Path = patientWeightTag.GetPath(), VR = patientWeightTag.GetDefaultVR().Code, Level = QueryTagLevel.Instance },
+                    new AddExtendedQueryTagEntry { Path = patientAgeTag.GetPath(), VR = patientAgeTag.GetDefaultVR().Code, Level = QueryTagLevel.Instance },
+                });
             Assert.Equal(OperationRuntimeStatus.Completed, operationStatus.Status);
 
+            // Check specific tag
+            GetExtendedQueryTagEntry actual;
+
+            actual = await _tagManager.GetTagAsync(patientWeightTag.GetPath());
+            Assert.Equal(patientWeightTag.GetPath(), actual.Path);
+            Assert.Null(actual.Errors);
+
+            actual = await _tagManager.GetTagAsync(patientAgeTag.GetPath());
+            Assert.Equal(patientAgeTag.GetPath(), actual.Path);
+            Assert.Equal(2, actual.Errors.Count);
+
+            // Query multiple tags
+            var multipleTags = (await _tagManager.GetTagsAsync(2, 0)).ToList();
```

Comment:
I feel this may lead to flaky test, but let's see -- maybe we can fix other tests who pollute  environment instead.

Reply:
I wrote the tests to only fetch the number of tags I added. So if there are more tags than expected, we'll simply fetch the first 2 tags (whatever they are), and ensure we get the same values. That will only work though as long as tests for extended query tags run sequentially, which will be the case as long as they're in the same test class/collection